src/app/zeko/da-layer/contracts/State.sol

```diff
@@ -75,9 +75,14 @@ abstract contract State {
      * @dev Mapping of batchId to RollupBatch, where batchId is the Poseidon
      * hash of the batch fields.
      */
-    mapping(bytes32 => RollupBatch) internal batches;
+    mapping(uint256 => RollupBatch) public batches;
+    uint256 public batchesLength = 0;
 
-    address public sequencer;
+    bool public genesisInitialized = false;
+    string public genesisState;
+
+    mapping(HashedMinaPublicKey => mapping(uint256 => bool)) validatorSigned;
 
-    bytes32 public lastBatchId;
+    // we allow only sequencer to post batches for now
+    address public sequencer;
```

Comment:
I don't think you need this. If we wanted only a single sequencer to be able to post to the DA, we would simply limit the access to the DA chain itself to only the sequencer, rather than doing it in a smart contract.

Unless I'm misunderstanding something?

Reply:
oh yeah we don't need it anymore, we needed it before we've had the location in the outer account